src/localizePropsDecorator.js

```diff
@@ -30,30 +38,46 @@ export default function localizePropsDecorator({ containerNamespace, local }) {
       static _instanceCount = 0;
 
       globalizeLocalState = obj => reduce(obj, (acc, v, k) => {
-        const globalKey = local[k] ? `${containerNamespace}${k}` : k;
+        const globalKey = `${containerNamespace}${k}`;
+
+        const { ___local = {} } = this.props;
+
         return {
           ...acc,
-          [globalKey]: local[k] ? {
-            ...this.props[globalKey],
+          [globalKey]: {
+            ...___local[globalKey],
             [this.instanceNamespace]: v
-          } : v
+          }
         };
       }, {});
 
-      localizeLocalState = obj => reduce(obj, (acc, v, k) => {
-        const localKey = k.replace(containerNamespace, '');
-        return {
-          ...acc,
-          [localKey]: local[localKey] && v ? v[this.instanceNamespace] : v
-        };
-      }, {});
+      localizeLocalState = obj => {
+        const state = reduce(obj, (acc, v, k) => {
+          const localKey = k.replace(containerNamespace, '');
+          return {
+            ...acc,
+            [localKey]: local[localKey] && v ? v[this.instanceNamespace] : v
+          };
+        }, {});
+        return state;
+      }
 
-      localizeProps = props => this.localizeLocalState({
+      localizeProps = ({ ___local = {}, ...props }) => ({
         ...props,
+        ...this.localizeLocalState(pick(___local, globalizedLocalKeys)),
         transition: (...args) => {
           if (args.length === 1 && t.Object.is(args[args.length - 1])) {
```

Comment:
yes this `args.length - 1` is very useless, can you switch to just `args[0]` here too?